### DISCUSSION TITLE
I am not able to activate Microsoft.Extensions.Logging.Console and a file export

### DISCUSSION THREAD

**Discussion by Kuwito** [fo-dicom/fo-dicom#1627] [Q&A] [answered]
The first thing I want to do is to apologize if my knowledge is limited.

I am trying to activate the loggin using Microsoft.Extensions.Logging, but I am not able to.

If I use: 
// create a logger factory
var loggerFactory = LoggerFactory.Create(
    builder => builder
                // add console as logging target
                .AddConsole()
                // add debug output as logging target
                //.AddDebug()
                // set minimum level to log
                .SetMinimumLevel(Microsoft.Extensions.Logging.LogLevel.Debug)
);
// create a logger
var logger = loggerFactory.CreateLogger("programa");
logger.LogInformation("This is log message.");

I see the line correctly, but I don't see the fo-dicom debug.

To see the fo dicom debug, I put this line: 
LogManager.SetImplementation(ConsoleLogManager.Instance);

But although it is showing it in debug, I don't know if it is using Microsoft.Extensions.Logging or not...

The thing is that I would like to be able to send the loggin to an external file and I don't really know how to do it.

If it's no bother, can someone help me a little?
Thanks in advance for the help and sorry for my english level ;-)

**Comment by amoerie** [comment, marked as answer]
To make Fellow Oak DICOM use Microsoft.Extensions.Logging, you need to use dependency injection.

If you are using the host builder from ASP.NET Core, you can do this:

```csharp
var host = Host.CreateDefaultBuilder(args)
    .ConfigureLogging(logging =>
    {
        logging.ClearProviders();
        logging.AddConsole();
        logging.SetMinimumLevel(LogLevel.Information);
    })
    .ConfigureServices(services =>
    {
        services.AddFellowOakDicom();
        services.AddHostedService<Worker>();
        // This will be injected into the DICOM service
        services.AddSingleton<CustomDependency>();
    })
    .Build();

// This is still necessary for now, but we want to get rid of this in the long term
DicomSetupBuilder.UseServiceProvider(host.Services);

host.Run();
```

If you are not using the host builder, you'll need to make your own service collection:

```csharp
var services = new ServiceCollection();
services.AddLogging(logging =>
{
    logging.ClearProviders();
    logging.AddConsole();
    logging.SetMinimumLevel(LogLevel.Information);
});
services.AddFellowOakDicom();
var serviceProvider = services.BuildServiceProvider();
DicomSetupBuilder.UseServiceProvider(serviceProvider);
```

**Reply by Kuwito** [reply]
Thank you very much for your help.
It has helped me a lot. It is what I needed!
